crates/opte-api/src/ip.rs

```diff
@@ -94,68 +94,71 @@ impl Display for IcmpEchoReply {
-///
-/// Respond to a cilent's Discover and Request messages with Offer+Ack
-/// replies based on the information contained in this struct.
-///
-/// XXX Currently we return the same options no matter what the client
-/// specifies in the parameter request list. This has worked thus far,
-/// but we should come back to this and comb over RFC 2131 more
-/// carefully -- particularly §4.3.1 and §4.3.2.
-pub struct DhcpAction {
-    /// The client's MAC address.
-    pub client_mac: MacAddr,
-
-    /// The client's IPv4 address. Used to fill in the `yiaddr` field.
-    pub client_ip: Ipv4Addr,
-
-    /// The client's subnet mask specified as a prefix length. Used as
-    /// the value of `Subnet Mask Option (code 1)`.
-    pub subnet_prefix_len: Ipv4PrefixLen,
-
-    /// The gateway MAC address. The use of this action assumes that
-    /// the OPTE port is acting as gateway; this MAC address is what
-    /// the port will use when acting as a gateway to the client. This
-    /// is used as the Ethernet header's source address.
-    pub gw_mac: MacAddr,
-
-    /// The gateway IPv4 address. This is used for several purposes:
-    ///
-    /// * As the IP header's source address.
-    ///
-    /// * As the value of the `siaddr` field.
-    ///
-    /// * As the value of the `Router Option (code 3)`.
-    ///
-    /// * As the value of the `Server Identifier Option (code 54)`.
-    pub gw_ip: Ipv4Addr,
-
-    /// The value of the `DHCP Message Type Option (code 53)`. This
-    /// action supports only the Offer and Ack messages.
-    pub reply_type: DhcpReplyType,
+/// Per-guest DHCP options.
+#[derive(Debug, Default, Clone, Serialize, Deserialize)]
+pub struct DhcpCfg {
+    /// Hostname to assign connected guest over DHCP.
+    pub hostname: Option<DomainName>,
 
-    /// A static route entry, sent to the client via the `Classless
-    /// Static Route Option (code 131)`.
-    pub re1: SubnetRouterPair,
+    /// Local domain of connected guest over DHCP.
+    pub host_domain: Option<DomainName>,
 
-    /// An optional second entry (see `re1`).
-    pub re2: Option<SubnetRouterPair>,
-
-    /// An optional third entry (see `re1`).
-    pub re3: Option<SubnetRouterPair>,
-
-    /// An optional list of 1-3 DNS servers.
-    pub dns_servers: Option<[Option<Ipv4Addr>; 3]>,
-
-    /// A list of domain names used when resolving relative names via
-    /// DNS.
-    pub domain_list: Vec<DomainName>,
-}
+    /// A list of domain names used during DNS resolution.
+    ///
+    /// Resolvers will use the provided list when resolving relative domain
+    /// names.
+    pub domain_search_list: Vec<DomainName>,
+
+    /// IPv4 external DNS servers provided to a guest.
+    pub dns4_servers: Vec<Ipv4Addr>,
+
+    /// IPv6 external DNS servers provided to a guest.
+    pub dns6_servers: Vec<Ipv6Addr>,
+}
+
+impl DhcpCfg {
+    /// Provide DNS servers which allow basic name resolution via CloudFlare/Google.
+    pub fn base_reachable() -> DhcpCfg {
```

Comment:
Mentioned in another comment, but I'd remove this from opte proper and leave it to consumers to decide on a default.